### PR TITLE
Add tests to validate posts to the module route 

### PR DESCRIPTION
**Pull request by dhollinger**
These tests will test the get routes for `/` and `/heartbeat`. For now, I will be redirecting my focus on testing the helper methods rather than the post routes first. Those methods are more important to test rather than the post routes. 

Tests for the post routes will resume after the helpers are tested.

### CODE REVIEW

File: spec/unit/puppet_webhook_spec.rb [hunk 1]
```diff
@@ -1,9 +1,24 @@
...
     get '/heartbeat'
     expect(last_response).to be_ok
     expect(last_response.body).to eq('{"status":"success","message":"running"}')
   end
+
+  context 'github module endpoint' do
+    it '/module returns ok' do
+      stub_const('LOGGER', Logger.new('spec/fixtures/test.log'))
+      basic_authorize('puppet', 'puppet')
+      post '/module', { repository: { name: 'puppet-module'} }.to_json, { 'Content-Type' => 'application/json', 'X-Github-Event' => 'push' }
```
Comment (alexjfisher) [inline spec/unit/puppet_webhook_spec.rb]: You can also set headers this way.
```ruby
header 'Content-Type', 'application/json'
header 'X-GitHub-Event', 'push'
post '/module', { repository: { name: 'puppet-module'} }.to_json
```

Comment (ekohl) [reply spec/unit/puppet_webhook_spec.rb]: And this could use some assertion about the response.

Comment (alexjfisher) [reply spec/unit/puppet_webhook_spec.rb]: The post body is missing `owner: { login: 'login' }` so the parser chokes and the route never gets called.

Comment (alexjfisher) [reply spec/unit/puppet_webhook_spec.rb]: Should we just use our github fixture?
```
post '/module', read_fixture('github/update.json')
```

File: spec/unit/puppet_webhook_spec.rb [hunk 2]
```diff
@@ -1,9 +1,26 @@
...
   end
+
+  context 'github module endpoint' do
+    it '/module returns ok' do
+      stub_const('LOGGER', Logger.new('spec/fixtures/test.log'))
+      basic_authorize('puppet', 'puppet')
+      header 'Content-Type', 'application/json'
+      header 'X-GitHub-Event', 'push'
+      post '/module', { repository: { name: 'puppet-module', owner: { login: 'login' } } }.to_json
+      expect(last_response).to be_ok
```
Comment (alexjfisher) [inline spec/unit/puppet_webhook_spec.rb]: I added this, otherwise the test now passes when it shouldn't (yet).

**Comment by alexjfisher** [comment]
@dhollinger I pushed a couple of commits.  Still needs work.  It's now blowing up with `uninitialized constant Deployments::COMMAND_PREFIX` here. https://github.com/voxpupuli/puppet_webhook/blob/c14fa359c55e5c94ec9f34161a482efefd998c66/lib/helpers/deployments.rb#L36

**Comment by binford2k** [comment]
This will have to be reworked now to match #40

**Pull request merged**